File: Python/pat1002.py
#! /usr/bin/env python
# coding:utf-8
"""

Author: zhiying
Date: 2019.7.31
Description: 关注志颖博客(www.zhouzying.cn),一起探讨爬虫技术！

"""


# split the input number.
def split_number(nums):
    i = 0
    item = {}
    if nums < 10:
        item[i] = nums
    else:
        while nums != 0:
            item[i] = nums % 10
            i += 1
            if nums // 10 != 0:
                nums = nums // 10
                if nums < 10:
                    item[i] = nums
                    nums = nums // 10
    return item


# make sum
def make_sum(splited_nums):
    sums = 0
    for j in range(len(splited_nums)):
        sums += splited_nums[j]
    return sums


def main():
    pinyin = {0: 'ling', 1: 'yi', 2: 'er', 3: 'san', 4: 'si', 5: 'wu', 6: 'liu', 7: 'qi', 8: 'ba', 9: 'jiu'}
    num = eval(input("请输入一个整数：\n"))
    item = split_number(num)
    all_sums = make_sum(item)
    output_nums = split_number(all_sums)
    for j in range(len(output_nums)-1, -1, -1):
        key = output_nums[j]
        if j == 0:
            print(pinyin[key], end='\n')
        else:
            print(pinyin[key], end=' ')


if __name__ == '__main__':
    main()

File: C/pat1002.c
/**
  pat1002
Author: zhiying
Date: 2020.3.10

*/

#include <stdio.h>

void split_numbers(int n,int *splited_numbers,int *len){

	int array_len;
	int i=0;
    if (n<10){
		splited_numbers[i] = n;
		
    }else{
		while(n != 0){
		
			splited_numbers[i] = n % 10;
			i++;
			n /= 10;
			if (n <10){
			
				splited_numbers[i] = n;
				n /= 10;
			}
		}
    }

	array_len = i + 1;
	
	*len = array_len;
}

void make_sum(int *splited_numbers, int len, int *all_sum){
	
	int numbers_sum=0;
	int i;

	for (i=0;i<len;i++){

		numbers_sum += splited_numbers[i];
		
	}

	*all_sum = numbers_sum;

}

void print_values(char *a[],int *b,int len){
	int i;
	int index;
	for(i=len-1;i>=0;i--){
	
		index = b[i];
		printf("%s",a[index]);
		if (i != 0){
			printf(" ");

		}
	}


}

int main(){

    char n[100];
    char *p;	
	p = n;
	char *all_sum[] = {"ling","yi","er","san","si","wu","liu","qi","ba","jiu"};
	int a[100];
	int b[100];
	int len=0;
	int sum_len;
	int result;
	scanf("%s",p);
	while(*p != '\0'){
	
		a[len] = (int)(*p) - 48;
		p++;
		len++;
	}
    make_sum(a,len,&result);
	split_numbers(result,b,&sum_len);
	print_values(all_sum,b,sum_len);

	return 0;
}
